WordPress/gutenberg
# List view: Allow selected block to override roving tabindex

## What?


Fixes #48337

## Why?


See above.

## How?


If a block is selected, I replace the roving tabindex value with 0. Roving tabindex only keeps track of the last focused block and has no way to know about blocks selected in the canvas.

## Testing Instructions





1. Open a post or page.
2. Insert some blocks, at least three.
3. Select the second block in the canvas.
4. Open the list view and notice that the second block is selected and focus is placed on it. This is because the `tabindex="0"` is set.
5. Select the third block in the canvas.
6. Press the shortcut to open the list view. On Windows, it is alt+shift+o.
7. Notice how focus is now placed on the third block.

### Testing Instructions for Keyboard


No difference.

## Screenshots or screencast 

## Notes

I know this is not a perfect solution due to the way things could get odd with the options button. In any case, the block will always receive focus but the correct options button will not. This is because the block will have 0 tabindex which would make it the first tabbable. You can better understand this by looking at `useFocusFirstElement` hook. I think this is a fair trade-off for the benefit of not having a large refactor. The other option is, I disable the roving tabindex provider with a custom parameter and handle this in list view that way being in block-editor, I can have awareness of selected blocks.
